src/lib/command.rs

```diff
@@ -200,6 +201,10 @@ pub(crate) fn run_command(command_string: &str, args: &Option<Vec<String>>, vali
 /// Runs the given task command.
 pub(crate) fn run(step: &Step) {
     let validate = !step.config.should_ignore_errors();
+    let level = env::var("CARGO_MAKE_INTERNAL_RECURSION")
+        .unwrap_or_else(|_| "0".to_string());
+
+    envmnt::set("CARGO_MAKE_INTERNAL_RECURSION", (level.parse::<i64>().unwrap() + 1).to_string());
```

Comment:
not sure this is the right place. this runs any command, not just cargo-make calling cargo-make.
so when cargo-make calls cargo build or test, it goes through here. so you are counting lots of stuff...
also if cargo-make is invoked from script, you missed it (in case of workspaces i generate a script per member).
so i think this logic needs to go to the cli.run at the very start, before any printing starts....